prometheus-ksonnet/grafana/datasources.libsonnet

```diff
@@ -38,29 +34,28 @@
   */
   grafana_add_datasource(name, url, default=false, method='GET')::
     configMap.withDataMixin({
-      ['%s.yml' % name]: $.util.manifestYaml($.grafana_datasource(name, url, default, method)),
+      ['%s.yml' % name]: $.util.manifestYaml({
+        apiVersion: 1,
+        datasources: [$.grafana_datasource(name, url, default, method)],
+      }),
     }),
 
   // Generates yaml string containing datasource config
-  grafana_datasource_with_basicauth(name, url, username, password, default=false, method='GET')::
-    {
-      apiVersion: 1,
-      datasources: [{
-        name: name,
-        type: 'prometheus',
-        access: 'proxy',
-        url: url,
-        isDefault: default,
-        version: 1,
-        editable: false,
-        basicAuth: true,
-        basicAuthUser: username,
-        basicAuthPassword: password,
-        jsonData: {
-          httpMethod: method,
-        },
-      }],
+  grafana_datasource_with_basicauth(name, url, username, password, default=false, method='GET', type='prometheus'):: {
```

Comment:
Can't this call `grafana_datasource()` and mixin the auth items?

Reply:
that comes next. One thing at a time.